engine/src/main/java/org/hibernate/validator/internal/engine/ValidationContext.java

```diff
@@ -348,10 +363,22 @@ public void addConstraintFailures(Set<ConstraintViolation<T>> failingConstraintV
 	}
 
 	public boolean hasMetaConstraintBeenProcessed(Object bean, Path path, MetaConstraint<?> metaConstraint) {
+		// this is only useful if the constraint is defined for more than 1 group as in the case it's only
+		// defined for one group, there is no chance it's going to be called twice.
+		if ( metaConstraint.getGroupList().size() <= 1 ) {
```

Comment:
maybe we can replace this with `MetaConstraint#isInMultipleGroups()` we could cache that value as well?

Reply:
Good idea. Will do and force push.

Reply:
Done. It doesn't change anything though but it's cleaner anyway.